gateway/reverse_proxy.go

```diff
@@ -91,7 +89,13 @@ func urlFromService(spec *APISpec) (*apidef.HostList, error) {
 			return spec.LastGoodHostList, nil
 		}
 
-		ServiceCache.Set(spec.APIID, data, cache.DefaultExpiration)
+		if spec.Proxy.ServiceDiscovery.UseDiscoveryService {
```

Comment:
Is this check really needed? from what I can see `urlFromService ` is only called for Service discovery

Reply:
Probablly not, but the issue is that in other places it's conditional around the same servicecache global; we don't have a true control point for the feature so this will get refactored away at some point.

Reply:
I removed the flag on a rethink, slightly refactored code depending if cache is configured in the api spec.